diff --git a/common/c_cpp/src/c/queue.c b/common/c_cpp/src/c/queue.c
index 4701a78..027429c 100644
--- a/common/c_cpp/src/c/queue.c
+++ b/common/c_cpp/src/c/queue.c
@@ -265,6 +265,10 @@ wombatQueue_dispatchInt (wombatQueue queue, void** data, void** closure,
  
     /* remove the item */
     head                   = impl->mHead.mNext;
+    if (head == &impl->mTail) {
+       wthread_mutex_unlock (&impl->mLock);
+       return WOMBAT_QUEUE_OK;
+    }
     WQ_REMOVE (impl, head);
 
     if (data)
